src/plugins/controls/public/options_list/components/options_list_control.tsx

```diff
@@ -162,7 +162,10 @@ export const OptionsListControl = ({
         anchorPosition="downCenter"
         initialFocus={'[data-test-subj=optionsList-control-search-input]'}
         className="optionsList__popoverOverride"
-        closePopover={() => optionsList.dispatch.setPopoverOpen(false)}
+        closePopover={() => {
+          optionsList.dispatch.setPopoverOpen(false);
+          optionsList.publishNewSelections();
```

Comment:
Nice! 🎉 Tested this, and it seems to work as expected.

I wonder - should we do the same for the erase button? In my head, it makes sense that clicking this should **also** publish the new selections instantly without any debounce. You can achieve this via adding the following to the `onClick` of the trash `EuiIcon` in `options_list_popover_action_bar.tsx`:

```typescript
onClick={() => {
   optionsList.dispatch.clearSelections({});
   optionsList.publishNewSelections();
}}
```

Reply:
SGTM. But now this opens another question. When I hit the "Reset" button on the dashboard, the reset is also debounced. Is that ok? If not, we might need to consider storing the selections in component state and debounce those changes before publishing to explicit input.

Reply:
> we might need to consider storing the selections in component state and debounce those changes before publishing to explicit input.

Hmmm.... This seems like it might be the right call to me 🤔 In theory, if something is changing explicit input directly (as the reset button on dashboard does), we probably want that to happen instantly. I played around a bit, and it definitely feels awkward that the Dashboard reset button has such a delay, especially if you are **only** resetting control selections.

This would add some duplicated information, but I'm not sure that's a big deal - I doubt people are storing hundreds of selections in their selection array. Do you have thoughts about this @ThomThomson?

Reply:
I might need to dig in further before I fully understand this problem, but it seems to me that storing the selections in the component state for a moment before publishing them isn't that big of a deal.

My main worry with this idea is that we might be overcomplicating the code here. From my current understanding, it seems like we should be able to separate the following two cases:

1. The user makes a selection in the popover - this should be debounced.
2. Any other way that a change to the selection is made... i.e. erase, reset etc - this should not be debounced.

Is it possible to do this debouncing in the popover component itself - or in the list component right on user interaction? We might have been on the wrong track by debouncing in the Options List Embeddable class...